sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala

```diff
@@ -122,6 +124,12 @@ private[hive] class SparkExecuteStatementOperation(
           timeFormatters)
       case _: ArrayType | _: StructType | _: MapType | _: UserDefinedType[_] =>
         to += toHiveString((from.get(ordinal), dataTypes(ordinal)), false, timeFormatters)
+      case YearMonthIntervalType =>
```

Comment:
How about nested types?

Reply:
If you mean the array, map types, they are converted to strings and they don't have a structure that could be recognized by Hive lib. Don't think I can do something here in the PR. This is a restriction of current implementation, and it should be solved in general way.

Reply:
For example, after the PR https://github.com/apache/spark/pull/32120 is merged:
```sql
0: jdbc:hive2://localhost:10000/default> select array(timestamp'2021-01-01 01:02:03.000001' - date'2020-12-31');
+----------------------------------------------------+
| array(subtracttimestamps(TIMESTAMP '2021-01-01 01:02:03.000001', DATE '2020-12-31')) |
+----------------------------------------------------+
| [1 01:02:03.000001000]                             |
+----------------------------------------------------+
```

Reply:
I mean the client-side get the plain YearMonthIntervalType/DayTimeIntervalType values via `HiveIntervalYearMonth/DayTimeIntervalType`.toString, but the ones that in nested types are handled in `toHiveString`. They look the same but these branches seem to be redundant and may cause inconsistency if the underlying hive changed

Is 
```scala
case YearMonthIntervalType | DayTimeIntervalType  | _: ArrayType | _: StructType | _: MapType | _: UserDefinedType[_] =>
```
better as we handle them consistently at spark-side?

Reply:
If I do as you propose, I get the exception on my test:
```
[info]   Cause: java.lang.ClassCastException: java.lang.String cannot be cast to org.apache.hadoop.hive.common.type.HiveIntervalDayTime
[info]   at org.apache.hive.service.cli.ColumnValue.toTColumnValue(ColumnValue.java:174)
[info]   at org.apache.hive.service.cli.RowBasedSet.addRow(RowBasedSet.java:60)
[info]   at org.apache.hive.service.cli.RowBasedSet.addRow(RowBasedSet.java:32)
```
at
```java
    case INTERVAL_YEAR_MONTH_TYPE:
      return stringValue((HiveIntervalYearMonth) value);
    case INTERVAL_DAY_TIME_TYPE:
      return stringValue((HiveIntervalDayTime) value);
```
Probably, I could just replace the casting by string casting. Let me try.

Reply:
done